src/websocket.js

```diff
@@ -121,6 +121,8 @@ class Websocket {
             errorString += ' - ' + text;
         }
         log.error(errorString);
+        // The stream error is sent to the Strophe error method, allowing it to be handled.
+        Strophe.error(errorString);
```

Comment:
Thanks for the PR. I understand the problem that you have, but this is not a proper fix.

The `Strophe` object isn't imported into this file (and shouldn't be to avoid circular dependencies) and you're relying on `Strophe` being a global here, which is not a guarantee.

For a proper fix, we'll need calls to `log.error` to somehow be surfaced towards the `Strophe` object so that they can be overridden or intercepted via `Strophe.error`.

I'm not sure right now what the right solution will be. Maybe the right solution is to just throw and event inside `log.error` (and `log.info` etc.) which is caught by the Strophe object.